File: src/main/java/decorator/StudentDiscount.java
package decorator;

public class StudentDiscount implements Discount {
    public static final double RATE = 0.03;
    public static final String HISTORY = "학생 할인";

    @Override
    public double amountAfterDiscount(double amount) {
        return amount * RATE;
    }

    @Override
    public String history() {
        return HISTORY;
    }
}

File: kotlin/src/main/kotlin/decorator/StudentDiscount.kt
package decorator

class StudentDiscount: Discount {
    companion object {
        const val RATE = 0.03
        const val HISTORY = "학생 할인"
    }

    override fun amountAfterDiscount(amount: Double): Double {
        return amount * RATE
    }

    override fun history(): String {
        return HISTORY
    }
}